api/app/Actions/EventType/DeleteEventTypeAction.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\EventType;
+
+use App\Exceptions\EventTypeNotFoundException;
+use App\Repositories\EventType\EventTypeRepository;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Support\Facades\Auth;
+
+final class DeleteEventTypeAction
+{
+    private $eventTypeRepository;
+
+    public function __construct(EventTypeRepository $eventTypeRepository)
+    {
+        $this->eventTypeRepository = $eventTypeRepository;
+    }
+
+    public function execute(DeleteEventTypeRequest $request): void
+    {
+        try {
+            $eventType = $this->eventTypeRepository->getById($request->getId());
```

Comment:
The exception should not be thrown, I will explain in comments for repository file.